multiroom_model/simulation.py

```diff
@@ -34,105 +41,242 @@ def __init__(self,
+        
+        # Make a result numpy matrix
+        result = np.zeros((size, size))
+
+        # For each aperture calculate the flux and add it to the resultant matrix
+        for c in self._aperture_calculators:
+            aperture_calculator, room1_index, room2_index, _, _ = c
+            is_outdoor_aperture = room2_index is None
+            i = room1_index+1
+            j = 0 if is_outdoor_aperture else room2_index+1
+            f = aperture_calculator.trans_matrix_contributions(wind_speed, wind_direction_in_radians)
+            result[i, j] += f.from_1_to_2
+            result[j, i] += f.from_2_to_1
+
+        return result
+
     @staticmethod
-    def apply_window_results(room_results, window_results):
-        return room_results
+    def apply_aperture_results(room_results, aperture_results, solved_time):
+        """
+        Applies the effect of the aperture results, to alter the state of the rooms
+        Return the new room concentrations at the final time
+        """
+        # Make a new result from the current result the the solved time
+        result = [result.loc[[solved_time], :].astype(float) for result in room_results]
+        # Go through all the aperture results
+        for room_1_concentration_change, room_2_concentration_change, room1_index, room2_index in aperture_results:
+            # Adjust the concentrations of room_1 int the new results
+            new_room_1_value = result[room1_index].loc[solved_time, :].add(
+                room_1_concentration_change, fill_value=0.0)
+            result[room1_index].loc[solved_time, :] = new_room_1_value
+            # If there is a room 2, adjust the concentrations of room_2 int the new results
+            if (room2_index is not None):
+                new_room_2_value = result[room2_index].loc[solved_time, :].add(
+                    room_2_concentration_change, fill_value=0.0)
+                result[room2_index].loc[solved_time, :] = new_room_2_value
+        # return the augmented results
+        return result
 
     @staticmethod
     def build_room_evolver_starmap(room, global_settings):
+        """
+        Create one room evolver
+        """
         return RoomInchemPyEvolver(room, global_settings)
 
     @staticmethod
     def run_room_evolver_starmap_txt(evolver, t0, t_interval, initial_text_file):
+        """
+        Use the room evolver to calculate new room concentrations
+        Start with an initial text file of concentrations
+        """
         df, _ = evolver.run(t0=t0, seconds_to_integrate=t_interval, initial_text_file=initial_text_file)
         return df
 
     @staticmethod
     def run_room_evolver_starmap(evolver, t0, t_interval, initial_dataframe):
+        """
+        Use the room evolver to calculate new room concentrations
+        Start with an initial dataframe of concentrations
+        """
         df, _ = evolver.run(t0=t0, seconds_to_integrate=t_interval, initial_dataframe=initial_dataframe)
         return df
```

Comment:
Do you think this needs to be two methods? Perhaps we could have one method that checks for 'dataframe' vs 'text' type? If it is left as two methods, could we alter the docstrings so they are different?